app/lib/features/calendar_sync/calendar_sync.dart

```diff
@@ -91,17 +92,54 @@ Future<void> initCalendarSync({bool ignoreRejection = false}) async {
           .listen(
     eventsToSyncProvider,
     (prev, next) async {
-      if (!next.hasValue) {
+      final events = next.valueOrNull;
+      if (events == null) {
         _log.info('ignoring state change without value');
         return;
       }
-      // FIXME: we probably want to debounce this ...
-      await _refreshCalendar(calendarId, next.valueOrNull ?? []);
+      _scheduleRefresh(calendarId, events);
     },
     fireImmediately: true,
   );
 }
 
+Timer? _debounce;
+(String, List<EventAndRsvp>)? _next;
+bool _running = false;
+
+// schedules an update of the calender
+// makes sure there is only one running at a time
+void _scheduleRefresh(
+  String calendarId,
+  List<EventAndRsvp> events,
+) {
+  _debounce?.cancel(); // cancel the current debounce;
+  _debounce = Timer(const Duration(seconds: 3), () {
+    _debounce = null;
+    _refreshLoop();
+  });
+  _next = (calendarId, events);
+}
+
+Future<void> _refreshLoop() async {
+  if (_running) return;
+  _running = true;
+  try {
+    while (true) {
+      final next = _next;
+      _next = null; // clear it
+      if (next == null) {
+        break;
+      }
+
+      final (calendarId, events) = next;
+      await _refreshCalendar(calendarId, events);
+    }
+  } finally {
+    _running = false;
+  }
+}
```

Comment:
when the loop is triggered it first ensure with the `_running`-flag that it isn't running. if it still is, this run quits as the other loop will take over.

during the loop we take the currently stored params and execute the refreshCalendar with it, once at a time. We reset the globally stored params at the start of the loop, but if an update happens while we are running, the next loop cycle will pull it up - giving us a super simple "latest-only" worker-queue system. If there are no more latest to process it quits and resets the `_running` flag to allow the next loop to start...